static/app/components/surfaceProvider.tsx

```diff
@@ -0,0 +1,18 @@
+import {createContext, type ReactNode, useContext} from 'react';
+
+export const SurfaceContext = createContext<string>('');
+
+export default function SurfaceProvider({
```

Comment:
Let's give this some jsdoc documentation. What is this used for, when should it be used, etc